src/hooks/useFetchData.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from "react";
+
+/**
+ * The useFetchData hook is used to fetch data from the backend. It returns the data as a json object.
+ * If not specified, the base url is set to 'http://localhost:3001/'.
+ * @param arg 
+ * @param base 
+ * @param callBack 
+ * @returns 
+ */
+const useFetchData = (arg: string, base?: string, callBack?: () => void) => {
```

Comment:
Which data is fetched from the backend? Maybe we should rename to useRunData?

Reply:
Yes that's currently a temporary hook. I am currently trying to figure out a more convienient version to fetch the data we need in a user friendly way..

I would be glad to get some inputs on figuring out what the best way to structure stateFiles' metadata is, for an efficient data fetching :-)